subprojects/solver/solver-smtlib/src/main/java/hu/bme/mit/theta/solver/smtlib/impl/generic/GenericSmtLibTermTransformer.java

```diff
@@ -254,6 +255,11 @@ public GenericSmtLibTermTransformer(final SmtLibSymbolTable symbolTable) {
             // Array
             put("select", exprArrayReadOperator());
             put("store", exprArrayWriteOperator());
+
+            // Proof
+            put("hyper-res", expectedFunc("hyper-res"));
+            put("asserted", expectedFunc("asserted"));
+            put("mp", expectedFunc("mp"));
```

Comment:
Are these generic across SMT-LIB, or only Horn specific? If the latter, it may be "nicer" to create a `GenericHornTermTransformer` that extends `GenericSmtLibTermTransformer`

Reply:
These are "generic" in the sense that all proofs (at least by Z3) are given using these predicates. I think because these are not Horn specific, but rather Z3 specific (and we know other solvers like to reuse Z3's conventions..), I'm inclined to keep them here. However, I do concede that right now, only CHC solving requires proof back-transformation, so I don't know if they should stay here just because in the future we might need them for other stuff.